src/main/java/haste/tasks/Deadline.java

```diff
@@ -0,0 +1,34 @@
+package haste.tasks;
+
+import haste.commands.Parser;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents a Deadline.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+
+    /**
+     * Creates a Deadline.
+     *
+     * @param description Task description.
+     * @param by End time of the Task.
+     * @param isComplete Completion status of Task.
+     */
+    public Deadline(String description, LocalDateTime by, boolean isComplete) {
+        super(description,isComplete);
+        this.by = by;
+    }
```

Comment:
Logical units within a block should be separated by one blank line - there's no line inbetween here and @Override.